numpy/core/_dtype.py

```diff
@@ -126,6 +126,9 @@ def _scalar_str(dtype, short):
         else:
             return "'%sU%d'" % (byteorder, dtype.itemsize / 4)
 
+    elif dtype.kind == '\x00':
+        return f"'{byteorder}{type(dtype).__name__}{dtype.itemsize * 8}'"
```

Comment:
Yes, I am fine with this, although don't have `_legacy` now to do this?

I am confused about this type of string, though, can't we use the `repr`?  `{byteorder}{TypeName}{itemsize}` doesn't need to be much of a reasonable repr?

Reply:
I'm fine with just using the repr here and in the other path, I agree that the bitsize isn't particularly useful and if someone thinks it's worth including they can include it in the repr. Mind if I take care of that in a followup?

Reply:
Sure, I am fine with that, since its all in flux somewhat alpha stage anyway.  Thanks @ngoldbaum lets just put it in.